index.js

```diff
@@ -15,6 +21,38 @@ app.get('/', (req, res) => {
   res.send('Hello World');
 });
 
+function construct_master(){
+	var master_products = all_products;
+	master_products.forEach(
+		(each_product) => each_product.categoryName = all_categories.filter(
+		// check if id of category element matches product categoryId
+			each_category => each_category.id === each_product.categoryId)[0].categoryName);
+	return master_products;
+}
+
+var product_master = construct_master();
+
+// endpoint to get all products
+app.get('/products/all', (req, res) => {
+  res.json(product_master);
+});
+
+// endpoint to get details of specific product
+app.get('/product/:id', (req, res) => {
+  var single_object = product_master.filter(
+    //filter out product which have specific id
+    each_product => each_product.id === req.params.id);
+  res.json(single_object);
+});
+
+// endpoint to get details of specific category
+app.get('/category/:ctyId', (req, res) => {
+  var category_objects =  product_master.filter(
```

Comment:
Use a more meaningful name e.g. `products_with_category` . And it's good practice to put this as its own function, that takes a `ctyId` as a parameter.